Sources/NetworkRequestHelper.swift

```diff
@@ -21,6 +21,7 @@ import XCTest
 ///    - ``MockNetworkService``
 ///    - ``RealNetworkService``
 class NetworkRequestHelper {
+    private let queue: DispatchQueue = .init(label: "com.adobe.testutils.networkrequesthelper.queue")
```

Comment:
Looks like you need to clean up the access control for these variables.
First, can these be `let` instead of `var` so they don't get reset.
Second, `orderedNetworkRequests` and `networkResponses` have public getters which need to be guarded by the DispatchQueue you created when their values are requested directly. You'll probably need to pass back a copy of the dictionaries so they don't get modified by the caller.  The variable `networkResponses` does have its own getter function, so it might be possible to simply make it private.

Reply:
I have updated based on your review. Let me know your thoughts.